modern/src/runtime/System.ts

```diff
@@ -470,24 +473,20 @@ class System extends MakiObject {
     return;
   }
 
-  strmid(str: string, start: number, len: number) {
-    unimplementedWarning("strmid");
-    return;
+  strmid(str: string, start: number, len: number): string {
+    return str.substring(start, start + len);
   }
 
-  strleft(str: string, nchars: number) {
-    unimplementedWarning("strleft");
-    return;
+  strleft(str: string, nchars: number): string {
+    return str.substring(nchars);
   }
 
-  strright(str: string, nchars: number) {
-    unimplementedWarning("strright");
-    return;
+  strright(str: string, nchars: number): string {
+    return str.substring(str.length - nchars);
   }
 
-  strsearch(str: string, substr: string) {
-    unimplementedWarning("strsearch");
-    return;
+  strsearch(str: string, substr: string): number {
+    return str.indexOf(substr);
```

Comment:
Do we know how Maki handles substrings that it can't find? I assume it returns `-1` like JavaScript or other similar languages, but it would be good to confirm. Manually for now would be fine, but with a test eventually.

Reply:
Yea, like everything else, we should definitely test them for real.

I'll include the doc string from MAKI though:
` Search a string for any occurance of substring. If the substring was
 found in the string, it will return the position of the substring in
 the string searched. If the substring is not found, the return value
 is -1.`